### DISCUSSION TITLE
Termux aiogram install

### DISCUSSION THREAD

**Discussion by syradeveloper** [aiogram/aiogram#1529] [Q&A] [answered]
Hello everyone, I have a fully functional bot on aiogram and I decided to host it from my phone. But when trying to install aiogram via termux, an error occurs at the pydantic_core (maturin) package installation stage.
I have rust, make, clang and everything else installed to build packages successfully. but in the end, the error contains
 
```
error occurred: Command ZERO_AR_DATE="1" "aarch64-linux-android-ar" "cq" "/data/data/com.termux/files/usr/tmp/pip-install-bsz9xlg5/maturin_d3417f1ce307464f8103f3bfde1b1ade/target/release/build/psm-6c7b6d9bde765414/out/libpsm_s.a" "/data/data/com.termux/files/usr/tmp/pip-install-bsz9xlg5/maturin_d3417f1ce307464f8103f3bfde1b1ade/target/release/build/psm-6c7b6d9bde765414/out/e4479bd7784abb10-aarch_aapcs64.o" with args "aarch64-linux-android-ar" failed to start: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }
```

# Does anyone have any idea how to fix this? Only if you do not take into the root rights on the device, I don't have root right

**Comment by syradeveloper** [comment, marked as answer]
# FIXED!

The problem was solved by installing the arch distribution on termux via proot-distro and a additional installation of pydantic_core and the library that created the error (forgot the name) via 
```
pacman -S python-pydantic-core python-(problematic-library-name)
```

**Comment by yojan21232e** [comment]
Greetings from Peru! 😊

The issue arises because aiogram is not compatible with Python 3.12. However, it works perfectly with earlier Python versions, such as Python 3.10. Below are the steps to resolve this issue by installing an older Python version in Termux:

Solution: Install Python 3.10 in Termux
Update Termux:
pkg update && pkg upgrade

Install necessary dependencies:
pkg install clang cmake ninja rust make

Install the tur-repo:
pkg install tur-repo

Install Python 3.10:
pkg install python3.10

Upgrade pip and wheel:
pip3.10 install --upgrade pip wheel

Usage: Install and run the bot
Clone the bot repository:
git clone <repository-link>

Navigate to the bot directory:
cd <repository-name>

Install the required dependencies:
pip3.10 install -r requirements.txt

Run the bot:
python3.10 main.py

Following these steps, you can successfully install and run the bot on Termux. If you're using Python 3.12, note that aiogram is not compatible, so you must switch to Python 3.10 or an earlier version.

Hope this helps! 😊
